### PR TITLE
[sgd] HuggingFace GLUE Fine-tuning Example

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Ports the Huggingface example for Glue fine tuning over.

Here are some removed features:
1. Tensorboard logging (should have a solution for this)
2. Loading from checkpoint automatically is not tested.
3. Does grad accumulation work still?
4. Saving/logging within epoch

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/util/sgd/torch/examples/transformers/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    # Install Apex
+    - git clone https://github.com/NVIDIA/apex;
+      cd apex &&
+      pip install -v --no-cache-dir  ./ ||
+      true
+
+
+file_mounts: {
+    # This should point to ray/python/ray/util/sgd.
+    /home/ubuntu/anaconda3/lib/python3.6/site-packages/ray/util/sgd/: ../../../,
```
Comment (maximsmol) [inline python/ray/util/sgd/torch/examples/transformers/cluster.yaml:76]: Probably shouldn't be in examples? We always removed this is the past

File: python/ray/util/sgd/torch/examples/transformers/transformers_example.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
+# coding=utf-8
+# This is a modified example originally from The Google AI Language Team
```
Comment (maximsmol) [inline python/ray/util/sgd/torch/examples/transformers/transformers_example.py:2]: I think you have to specify the person/group who modified the code (as per Apache, IANAL tho)

**Review by maximsmol — APPROVED** [review]
lgtm

File: python/ray/util/sgd/torch/examples/transformers/transformers_example.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+# Authors and The HuggingFace Inc. team.
```
Comment (richardliaw) [inline python/ray/util/sgd/torch/examples/transformers/transformers_example.py:3]:
```suggestion
# Authors and The HuggingFace Inc. team.
# Modified by Richard Liaw.
```

File: python/ray/util/sgd/torch/examples/transformers/cluster.yaml (hunk 1 above)
Comment (richardliaw) [inline python/ray/util/sgd/torch/examples/transformers/cluster.yaml:76]:
```suggestion
```

Comment (richardliaw) [reply python/ray/util/sgd/torch/examples/transformers/cluster.yaml]: done

File: python/ray/util/sgd/torch/examples/transformers/transformers_example.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/util/sgd/torch/examples/transformers/transformers_example.py:2]: done.

**Review by edoakes — APPROVED** [review]

**Pull request merged**